src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -124,15 +135,33 @@ def read_nwbfile(
     elif stream_mode == "ros3":
         import h5py
 
+        assert file_path is not None, "file_path must be specified when using stream_mode='ros3'"
+
         drivers = h5py.registered_drivers()
         assertion_msg = "ROS3 support not enbabled, use: install -c conda-forge h5py>=3.2 to enable streaming"
         assert "ros3" in drivers, assertion_msg
         io = NWBHDF5IO(path=file_path, mode="r", load_namespaces=True, driver="ros3")
 
-    else:
+    elif stream_mode == "remfile":
+        import remfile
+        import h5py
+
+        assert file_path is not None, "file_path must be specified when using stream_mode='remfile'"
+        rfile = remfile.File(file_path)
+        h5_file = h5py.File(rfile, "r")
+        io = NWBHDF5IO(file=h5_file, mode="r", load_namespaces=True)
+
+    elif file_path is not None:
         file_path = str(Path(file_path).absolute())
         io = NWBHDF5IO(path=file_path, mode="r", load_namespaces=True)
 
+    else:
+        import h5py
+
+        assert file is not None, "Unexpected, file is None"
```

Comment:
in this case, we should set JSON serializable to False

Reply:
Added a commit for this.